src/main.rs

```diff
@@ -39,31 +44,18 @@ fn query_watchman(epoch_nanoseconds: u64) -> Fallible<()> {
         .context("Couldn't start watchman")?;
 
     {
-        let time_seconds = epoch_nanoseconds / 1_000_000_000;
-
-        let watchman_query = json!(
-            [
-                "query",
-                git_work_tree,
-                {
-                    "since": time_seconds,
-                    "fields": ["name"],
-                    "expression": [
-                        "not", [
-                            "allof",[
-                                "since",
-                                time_seconds,
-                                "cclock"
-                            ],
-                            [
-                                "not",
-                                "exists"
-                            ]
-                        ]
-                    ]
-                }
-            ]
-        );
+        // the token following `since` expression can be either epoch second as integer or a clock id as string
```

Comment:
What I had in mind was that `main()` would first do a `match` on the `version`, and then call the top level `query_watchman` for each version (keeping the implementation details of each within each version). Would you mind giving that a shot? I feel like it might be more idiomatic Rust than attempting to parse a value in a context where it isn't used (`token_value` and `epoch_seconds` are both in local scope here, only 1 is used).

Reply:
Thanks for pointing it out! Sorry I was lazy but yes I agree it's better to separate it out.
I've made the changes, but I didn't move those functions to main, because the output handling part is almost the same, except v2 returns clock id before file list. Duplicating the logic is not ideal IMO.

Note that v2 supports both epoch seconds and clock id, I used the same logic as the official perl script. (checking if first character of the token is `c`)

New to rust, please let me know anywhere I can improve. :)